velox/connectors/hive/storage_adapters/s3fs/S3Util.h

```diff
@@ -77,13 +77,46 @@ inline bool isS3File(const std::string_view filename) {
       isOssFile(filename) || isCosFile(filename) || isCosNFile(filename);
 }
 
+inline unsigned char FromHex(unsigned char x) {
+  unsigned char y;
+  if (x >= 'A' && x <= 'Z')
+    y = x - 'A' + 10;
+  else if (x >= 'a' && x <= 'z')
+    y = x - 'a' + 10;
+  else if (x >= '0' && x <= '9')
+    y = x - '0';
+  else
+    assert(0);
+  return y;
+}
+
+inline std::string UrlDecode(const std::string& str) {
+  std::string strTemp = "";
+  size_t length = str.length();
+  for (size_t i = 0; i < length; i++) {
+    if (str[i] == '+')
+      strTemp += ' ';
+    if (str[i] == ' ')
+      strTemp += '+';
+    else if (str[i] == '%') {
+      assert(i + 2 < length);
+      unsigned char high = FromHex((unsigned char)str[++i]);
+      unsigned char low = FromHex((unsigned char)str[++i]);
+      strTemp += high * 16 + low;
+    } else
+      strTemp += str[i];
+  }
+  return strTemp;
+}
+
 inline void getBucketAndKeyFromS3Path(
     const std::string& path,
     std::string& bucket,
     std::string& key) {
-  auto firstSep = path.find_first_of(kSep);
-  bucket = path.substr(0, firstSep);
-  key = path.substr(firstSep + 1);
+  const std::string decodedPath = UrlDecode(path);
```

Comment:
Is this a limitation of the AWS S3 SDK library? Can you point to any links related to this issue?